### PR TITLE
Return a failure instead of crashing if shape inference can not be run because of unraked operand types

### PR DESCRIPTION
**Pull request by jorickert**
Some  `computeShape` functions assume/rely on the operands being ranked.
This PR adds a check to them, so that they return `failure` in case this assumption does not hold, instead of asserting. This allows callers of this function to handle this failure

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
@jenkins-droid test this please

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM,

I verified a few, let me just ask just to be sure. The `return failure()` that are added here are all in the `computeShape(...)` functions, which is fine since we expect to have successfully run shape inference before, were potentially unranked values have been at least ranked.

Thanks for adding "graceful" errors, much appreciated.

**Comment by jorickert** [comment]
@AlexandreEichenberger Yes, the `return failure()` are all added in `computeShape` . This allows callers of this function to decide how to handle failures in case `computeShape` fails because the input is unranked. 

 Before this PR, the functions `computeShapeAndAssertOnFailure` and `computeShape` would behave the same in many cases, as `computeShape` would assert internally.
 
 It would be nice if you could merge this, as I do not have merge rights

**Comment by AlexandreEichenberger** [comment]
Will do, for some reason I don't have the usual button to update to the latest. do you mind doing this on your end, and then I will merge. I asked for you to have access right to run the CIs. After a PR or two, will give you merge permission too.

**Comment by jorickert** [comment]
@AlexandreEichenberger I rebased onto the latest main

**Comment by jorickert** [comment]
> **Jenkins Linux amd64** Build [#16064](https://www.onnxmlir.xyz/jenkinx/job/ONNX-MLIR-Pipeline-Docker-Build/16064/) [push] Return a failure instead... failed after 53 min

@AlexandreEichenberger 
The CI error looks like some issue with the CI setup itself to me:
`s = '[0.002s][warning][perf,memops] Cannot use file /tmp/hsperfdata_root/232111 because it is locked by another process (e...QAAAAABAAAAAQEAAAABAQEBAAABAAABAQABAAEAAAEBAAEBAQABAQABAAAAAQEBAQEAAAABAQEBAQEAAQAAAAAAAQEAAAEAAAABAQAAAQEAAQEBAAEA"}]'
`

**Pull request merged**
